sway/input/cursor.c

```diff
@@ -194,15 +366,54 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 		}
 	}
 
-	// reset cursor if switching between clients
-	struct wl_client *client = NULL;
-	if (surface != NULL) {
-		client = wl_resource_get_client(surface->resource);
-	}
-	if (client != cursor->image_client) {
+	// Handle cursor image
+	if (surface) {
+		// Reset cursor if switching between clients
+		struct wl_client *client = wl_resource_get_client(surface->resource);
+		if (client != cursor->image_client) {
+			wlr_xcursor_manager_set_cursor_image(cursor->xcursor_manager,
+					"left_ptr", cursor->cursor);
+			cursor->image_client = client;
+		}
+	} else if (c && container_is_floating(c)) {
+		// Try a floating container's resize edge
+		enum resize_edge edge = find_resize_edge(c, cursor);
+		if (edge == RESIZE_EDGE_NONE) {
```

Comment:
This can be a `switch`, extracted in a separate function that returns a cursor image name from a resize edge.